core/multivm_processor.go

```diff
@@ -0,0 +1,163 @@
+package core
+
+import (
+	"math/big"
+
+	"github.com/ethereumproject/sputnikvm-ffi/go/sputnikvm"
+	"github.com/ethereumproject/go-ethereum/common"
+	"github.com/ethereumproject/go-ethereum/core/state"
+	"github.com/ethereumproject/go-ethereum/core/types"
+	evm "github.com/ethereumproject/go-ethereum/core/vm"
+	"github.com/ethereumproject/go-ethereum/crypto"
+	"github.com/ethereumproject/go-ethereum/logger"
+	"github.com/ethereumproject/go-ethereum/logger/glog"
+)
+
+var (
+	UseSputnikVM = false
+)
+
+func ApplyMultiVmTransaction(config *ChainConfig, bc *BlockChain, gp *GasPool, statedb *state.StateDB, header *types.Header, tx *types.Transaction, totalUsedGas *big.Int) (*types.Receipt, evm.Logs, *big.Int, error) {
+	tx.SetSigner(config.GetSigner(header.Number))
+
+	from, err := tx.From()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	vmtx := sputnikvm.Transaction {
+		Caller: from,
+		GasPrice: tx.GasPrice(),
+		GasLimit: tx.Gas(),
+		Address: tx.To(),
+		Value: tx.Value(),
+		Input: tx.Data(),
+		Nonce: new(big.Int).SetUint64(tx.Nonce()),
+	}
+	vmheader := sputnikvm.HeaderParams {
+		Beneficiary: header.Coinbase,
+		Timestamp: header.Time.Uint64(),
+		Number: header.Number,
+		Difficulty: header.Difficulty,
+		GasLimit: header.GasLimit,
+	}
+
+	current_number := header.Number
+	homestead_fork := config.ForkByName("Homestead")
+	eip150_fork := config.ForkByName("GasReprice")
+	eip160_fork := config.ForkByName("Diehard")
+
+	var vm *sputnikvm.VM
+	if eip160_fork.Block != nil && current_number.Cmp(eip160_fork.Block) >= 0 {
+		vm = sputnikvm.NewEIP160(&vmtx, &vmheader)
+	} else if eip150_fork.Block != nil && current_number.Cmp(eip150_fork.Block) >= 0 {
+		vm = sputnikvm.NewEIP150(&vmtx, &vmheader)
+	} else if homestead_fork.Block != nil && current_number.Cmp(homestead_fork.Block) >= 0 {
+		vm = sputnikvm.NewHomestead(&vmtx, &vmheader)
+	} else {
+		vm = sputnikvm.NewFrontier(&vmtx, &vmheader)
+	}
+
+Loop:
+	for {
+		ret := vm.Fire()
+		switch ret.Typ() {
+		case sputnikvm.RequireNone:
+			break Loop
+		case sputnikvm.RequireAccount:
+			address := ret.Address()
+			if statedb.Exist(address) {
```

Comment:
In Go there's a general convention to avoid `else` statements, preferring "breaking" patterns instead, so we could use
```go
		case sputnikvm.RequireAccountCode:
			address := ret.Address()
			if statedb.Exist(address) {
				vm.CommitAccountCode(address, statedb.GetCode(address))
                break
			}
			vm.CommitNonexist(address)
```
noting that `break` will break the switch, not the `for` `Loop`.

Reply:
It's only a style thing, though.

Reply:
Hmm, I'll fix this. But can you explain why? Code above is actually harder to read for me compared with an else statement.

Reply:
Fixed.

Reply:
I totally disagree with @whilei. This is perfectly legal use of `else` statement. They are typically omitted only when code itself returns/continues/breaks, etc., see [Effective Go](https://golang.org/doc/effective_go.html#if).
I think that `if/else` block is much more readable.

Reply:
Noted @tzdybal -- I definitely don't have my heart set on it, and think your take is valid; in fact I think it's more readable as Wei wrote it originally too. Whatever you want, @sorpaas